spring-integration-zip/src/main/java/org/springframework/integration/zip/splitter/UnZipResultSplitter.java

```diff
@@ -32,21 +33,37 @@
  * @since 1.0
  *
  */
-public class UnZipResultSplitter {
-
-	public List<Message<Object>> splitUnzippedMap(Map<String, Object> unzippedEntries) {
+public final class UnZipResultSplitter {
 
-		final List<Message<Object>> messages = new ArrayList<Message<Object>>(unzippedEntries.size());
+	public List<Message<Object>> splitUnzippedMap(Message<Map<String, Object>> message) {
+		return createMessages(message.getPayload(), message.getHeaders());
+	}
 
+	private List<Message<Object>> createMessages(Map<String, Object> unzippedEntries, MessageHeaders headers) {
+		List<Message<Object>> messages = createList(unzippedEntries.size());
 		for (Map.Entry<String, Object> entry : unzippedEntries.entrySet()) {
-			final String path = FilenameUtils.getPath(entry.getKey());
-			final String filename = FilenameUtils.getName(entry.getKey());
-			final Message<Object> splitMessage = MessageBuilder.withPayload(entry.getValue())
-					.setHeader(FileHeaders.FILENAME, filename)
-					.setHeader(ZipHeaders.ZIP_ENTRY_PATH, path).build();
-			messages.add(splitMessage);
+			messages.add(createMessage(entry, headers));
 		}
 		return messages;
 	}
 
+	private static List<Message<Object>> createList(int initialCapacity) {
+		return new ArrayList<Message<Object>>(initialCapacity);
+	}
```

Comment:
To be honest I don't see reason in this method, because of the single line of code.
Minor of course, but everything must have an explanation.
Would you mind to share your mind on the matter?